cli/pkg/update/update.go

```diff
@@ -0,0 +1,209 @@
+					stateEntry.LatestRelease.Version, currentVersion),
+				ReleaseInfo: &stateEntry.LatestRelease,
+			}, nil
+		}
+
+		return &UpdateInfo{
+			Update:  false,
+			Message: "Skip checking the latest version",
+		}, nil
+	}
+
+	// Check with latest release on github
+	info, err := LatestVersion(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = setVersionInfo(time.Now(), *info)
+	if err != nil {
+		return nil, err
+	}
+
+	v1, err := version.NewVersion(currentVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	v2, err := version.NewVersion(info.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	if v1.LessThan(v2) {
+		return &UpdateInfo{
+			Update: true,
+			Message: fmt.Sprintf("Latest version (%s) is greater than the current build version (%s)\n",
+				info.Version, currentVersion),
+			ReleaseInfo: info,
+		}, nil
+	}
+
+	return &UpdateInfo{
+		Update: false,
+		Message: fmt.Sprintf("Latest version (%s) is less than or equal to current build version (%s)",
+			info.Version, currentVersion),
+		ReleaseInfo: info,
+	}, nil
+}
+
+// This will fetch the latest version available for rill on github releases
+func LatestVersion(ctx context.Context) (*ReleaseInfo, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, Addr, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	req.Header.Set("Accept", "application/vnd.github.v3+json")
+
+	getToken := func() string {
+		if t := os.Getenv("GH_TOKEN"); t != "" {
+			return t
+		}
+		return os.Getenv("GITHUB_TOKEN")
+	}
+
+	if token := getToken(); token != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("token %s", token))
+	}
+
+	client := &http.Client{Timeout: time.Second * 15}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	out, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	success := resp.StatusCode >= 200 && resp.StatusCode < 300
+	if !success {
+		return nil, fmt.Errorf("error fetching latest release: %v", string(out))
+	}
+
+	var info *ReleaseInfo
+	err = json.Unmarshal(out, &info)
+	if err != nil {
+		return nil, err
+	}
+
+	return info, nil
+}
+
+type VersionInfo struct {
+	UpdateAt      time.Time   `yaml:"update_at"`
+	LatestRelease ReleaseInfo `yaml:"latest_release"`
+}
+
+func getVersionInfo() (*VersionInfo, error) {
+	content, err := dotrill.Get(StateFilename, VersionInfoStateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	var verionInfo VersionInfo
+	err = yaml.Unmarshal([]byte(content), &verionInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return &verionInfo, nil
+}
+
+func setVersionInfo(t time.Time, r ReleaseInfo) error {
+	data := VersionInfo{
+		UpdateAt:      t,
+		LatestRelease: r,
+	}
+
+	content, err := yaml.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	err = dotrill.Set(StateFilename, VersionInfoStateKey, string(content))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
It shouldn't serialize yaml here, it should just set two keys: `latest_version` and `latest_version_checked_at`